internal/webhook/v1/aerospikecluster_validating_webhook.go

```diff
@@ -733,206 +725,41 @@ func validateClusterSize(_ logr.Logger, sz int) error {
-	}
-
-	// network conf
-	networkConf, ok := config["network"].(map[string]interface{})
-	if !ok {
-		return fmt.Errorf(
-			"aerospikeConfig.network not a valid map %v", config["network"],
-		)
-	}
-
-	if err := validateNetworkConfig(networkConf, clientCert); err != nil {
-		return err
-	}
-
-	// namespace conf
-	nsListInterface, ok := config["namespaces"]
-	if !ok {
-		return fmt.Errorf(
-			"aerospikeConfig.namespace not a present. aerospikeConfig %v",
-			config,
-		)
-	} else if nsListInterface == nil {
-		return fmt.Errorf("aerospikeConfig.namespace cannot be nil")
-	}
-
-	if nsList, ok1 := nsListInterface.([]interface{}); !ok1 {
-		return fmt.Errorf(
-			"aerospikeConfig.namespace not valid namespace list %v",
-			nsListInterface,
-		)
-	} else if err := validateNamespaceConfig(
-		nsList, storage, clSize,
+	// It validates the aerospikeConfig schema and generic aerospikeConfig fields
+	if err := validation.ValidateAerospikeConfig(
+		aslog, version, configSpec.Value, clSize,
 	); err != nil {
 		return err
 	}
 
-	// logging conf
-	// config["logging"] is added in mutating webhook
-	loggingConfList, ok := config["logging"].([]interface{})
-	if !ok {
-		return fmt.Errorf(
-			"aerospikeConfig.logging not a valid list %v", config["logging"],
-		)
-	}
-
-	return validateLoggingConf(loggingConfList)
-}
-
-func validateLoggingConf(loggingConfList []interface{}) error {
-	syslogParams := []string{"facility", "path", "tag"}
-
-	for idx := range loggingConfList {
-		logConf, ok := loggingConfList[idx].(map[string]interface{})
-		if !ok {
-			return fmt.Errorf(
-				"aerospikeConfig.logging not a list of valid map %v", logConf,
-			)
-		}
-
-		if logConf["name"] != "syslog" {
-			for _, param := range syslogParams {
-				if _, ok := logConf[param]; ok {
-					return fmt.Errorf("can use %s only with `syslog` in aerospikeConfig.logging %v", param, logConf)
-				}
-			}
-		}
-	}
-
-	return nil
-}
-
-func validateNetworkConfig(networkConf map[string]interface{}, clientCert *asdbv1.AerospikeOperatorClientCertSpec,
-) error {
-	serviceConf, serviceExist := networkConf["service"]
-	if !serviceExist {
-		return fmt.Errorf("network.service section not found in config")
-	}
-
-	tlsNames := sets.Set[string]{}
-	// network.tls conf
-	if _, ok := networkConf["tls"]; ok {
-		tlsConfList := networkConf["tls"].([]interface{})
-		for _, tlsConfInt := range tlsConfList {
-			tlsConf := tlsConfInt.(map[string]interface{})
-			if tlsName, ok := tlsConf["name"]; ok {
-				tlsNames.Insert(tlsName.(string))
-			}
-
-			if _, ok := tlsConf["ca-path"]; ok {
-				if _, ok1 := tlsConf["ca-file"]; ok1 {
-					return fmt.Errorf(
-						"both `ca-path` and `ca-file` cannot be set in `tls`. tlsConf %v",
-						tlsConf,
-					)
-				}
-			}
-		}
-	}
-
-	if err := validateTLSClientNames(
-		serviceConf.(map[string]interface{}), clientCert,
-	); err != nil {
+	if err := validateNetworkConfig(configSpec.Value, clientCert); err != nil {
```

Comment:
can we rename functions here, like there are 2 functions `validateNetworkConfig` one in aerospikecluster_validating_webhook.go and another in validate.go, check others also.

Reply:
Actually I intentionality kept the names same to identify the similar functionality across diff pkg.
Eg: `validation.ValidateAerospikeConfig` is validation func for aerospikeConfig specific validation whereas `validateAerospikeConfig` in webhook covers both aerospikeConfig and AKO specific aerospikeConfig validation.
This way it's easier to track. 

Think of it like `validation` is a third party pkg and we are writing some wrapper funcs to add some AKO specific validations.

We can discuss and decide the renaming part if its still confusing.